central/policy/datastore/datastore_impl.go

```diff
@@ -241,12 +248,15 @@ func (ds *datastoreImpl) AddPolicy(ctx context.Context, policy *storage.Policy)
 	clonedPolicy.Categories = []string{}
 	err = ds.storage.Upsert(ctx, clonedPolicy)
 	if err != nil {
-		return clonedPolicy.Id, err
+		return "", ds.wrapWithRollback(ctx, tx, err)
 	}
 
 	err = ds.categoriesDatastore.SetPolicyCategoriesForPolicy(ctx, clonedPolicy.GetId(), policyCategories)
 	if err != nil {
-		return clonedPolicy.Id, err
+		return "", ds.wrapWithRollback(ctx, tx, err)
+	}
+	if err := tx.Commit(ctx); err != nil {
+		return "", ds.wrapWithRollback(ctx, tx, err)
```

Comment:
Did you test these to make sure the entire transaction rolled back if the category could not be set?